### PR TITLE
fix(lint-staged): markdownlint and prettier can race on md files

### PR DESCRIPTION
**Pull request by LeoMcA**
### Description

Prevent possible race condition between prettier and markdownlint in lint-staged, changed to use the order defined already in the package.json fix:md and lint:md commands.

I don't believe the other rules will race, since:
- `yarn test:front-matter-linter` doesn't appear to change the files, just runs jest on them
- prettier wouldn't run anyway on `*.{svg,png,jpeg,jpg,gif}`

@OnkarRuikar I don't seem to be able to assign you as a reviewer, but it would be great if you can take a look :)

### Additional details

https://github.com/okonet/lint-staged#task-concurrency

### Related issues and pull requests

Relates to https://github.com/mdn/content/pull/27936

### CODE REVIEW

**Review by queengooborg — APPROVED** [review]
This seems like a good solution to me, LGTM!

**Comment by queengooborg** [comment]
Argh, but I can't merge this because the two required tests aren't running, since modified files don't match their scope.

File: .lintstagedrc.json [hunk 1]
```diff
@@ -1,7 +1,8 @@
 {
-  "*": "prettier --ignore-unknown --write",
+  "!*.md": "prettier --ignore-unknown --write",
   "*.md": [
     "markdownlint-cli2-fix",
+    "prettier --write",
     "node scripts/front-matter_linter.js --fix true"
```
Comment (OnkarRuikar) [inline .lintstagedrc.json:6]:
```suggestion
    "markdownlint-cli2 --fix",
    "node scripts/front-matter_linter.js --fix true",
    "prettier --write"
```

a. The `markdownlint-cli2-fix` [is deprecated](https://github.com/DavidAnson/markdownlint-cli2#command-line). Use `--fix` option instead. We need to update package.json as well.
b. Recently we resolved clashes between Prettier and front-matter linter. For future proofing we always run Prettier last.

**Comment by OnkarRuikar** [comment]
> Argh, but I can't merge this because the two required tests aren't running, since modified files don't match their scope.

Anyone, please review and merge [this solution](https://github.com/mdn/content/pull/28277) fast.

**Review by bsmth — DISMISSED** [review]
There's some changes outstanding requested by @OnkarRuikar in https://github.com/mdn/content/pull/28306/files#r1280214197

**Pull request merged**